File: 프로그래머스/lv3/43238. 입국심사/입국심사.java
class Solution {
    
    static int N;
    
    public long solution(int n, int[] times) {
        long answer = 0;
        N = n;
        
        long L = 1, R = (long) Math.pow(10, 18);
        while(L < R) {
            long mid = (L + R) / 2;
            if(isAvailable(mid, times)) {
                R = mid;
                answer = mid;
            } else {
                L = mid + 1;
            }
        }
        
        return answer;
    }
    
    static boolean isAvailable(long x, int[] times) {
        long total = 0;
        for(int time : times) {
            total += x / (long) time;
        }
        
        return total >= (long) N;
    }
}

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    let answer = 0;
    
    let start = 1;
    let end = 1_000_000_000_000_000_000;
    
    while(true) {
        const mid = Math.floor((start + end) / 2);
        
        if(start >= end) break;
        
        if(isAvailable(mid, n, times)) {
            answer = mid;
            end = mid;
        } else {
            start = mid + 1;
        }
    }
    
    return answer;
}

const isAvailable = (mid, n, times) => {
    let sum = 0;
    
    for(let time of times) {
        sum += Math.floor(mid / time);
    }
    
    if(sum >= n) return true;
    
    return false;
};